src/super_gradients/training/datasets/pose_estimation_datasets/coco_pose_estimation_dataset.py

```diff
@@ -0,0 +1,212 @@
+
+import cv2
+import numpy as np
+import pycocotools
+from pycocotools.coco import COCO
+
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.transforms_factory import TransformsFactory
+from super_gradients.common.object_names import Datasets
+from super_gradients.common.registry.registry import register_dataset
+from super_gradients.training.datasets.data_formats.bbox_formats.xywh import xywh_to_xyxy, xyxy_to_xywh
+from super_gradients.training.datasets.pose_estimation_datasets.abstract_pose_estimation_dataset import AbstractPoseEstimationDataset
+from super_gradients.training.datasets.pose_estimation_datasets.coco_utils import (
+    CrowdAnnotationActionEnum,
+    remove_duplicate_annotations as remove_duplicate_annotations_fn,
+    remove_crowd_annotations,
+    remove_samples_with_crowd_annotations,
+)
+from super_gradients.training.samples import PoseEstimationSample
+from super_gradients.training.transforms.keypoint_transforms import AbstractKeypointTransform
+
+logger = get_logger(__name__)
+
+
+@register_dataset(Datasets.COCO_POSE_ESTIMATION_DATASET)
+class COCOPoseEstimationDataset(AbstractPoseEstimationDataset):
+    """
+    Dataset class for training pose estimation models using COCO format dataset.
+
+    Compatible datasets are
+    - COCO2017 dataset
+    - CrowdPose dataset
+    - Any other dataset that is compatible with COCO format
+
+    """
+
+    @resolve_param("transforms", TransformsFactory())
+    def __init__(
+        self,
+        data_dir: str,
+        images_dir: str,
+        json_file: str,
+        include_empty_samples: bool,
+        transforms: List[AbstractKeypointTransform],
+        edge_links: Union[List[Tuple[int, int]], np.ndarray],
+        edge_colors: Union[List[Tuple[int, int, int]], np.ndarray, None],
+        keypoint_colors: Union[List[Tuple[int, int, int]], np.ndarray, None],
+        remove_duplicate_annotations: bool = False,
+        crowd_annotations_action: Union[str, CrowdAnnotationActionEnum] = CrowdAnnotationActionEnum.NO_ACTION,
+    ):
+        """
+
+        :param data_dir:                     Root directory of the COCO dataset
+        :param images_dir:                   Path suffix to the images directory inside the dataset_root
+        :param json_file:                    Path suffix to the json file inside the dataset_root
+        :param include_empty_samples:        If True, images without any annotations will be included in the dataset.
+                                             Otherwise, they will be filtered out.
+        :param transforms:                   Transforms to be applied to the image & keypoints
+        :param edge_links:                   Edge links between joints
+        :param edge_colors:                  Color of the edge links. If None, the color will be generated randomly.
```

Comment:
We need to have these parameters because they are dataset-specific and stored as part of preprocessing parameters in the checkpoint metadata to enable .predict() function to work (Predict needs to know how to draw the pose with needs edge_links and edge_colors)